fix: refactor codebase and simplify it

armano2 opened the pull request:
further refactor started from #1

i removed support for legacy options and some additional options, all security errors are solved here

cc @slorber #4765

armano2 committed:
fix: refactor codebase

armano2 committed:
fix: simplify code base and remove unnecessary complexity

slorber committed:
Merge branch 'master' into refactor/simplify-code2

# Conflicts:
#	.gitignore
#	.travis.yml
#	package.json

slorber reviewed (approved):
Thanks @armano2 and sorry for the delay, will merge this 👍

slorber committed:
remove crawl from readme

slorber committed:
remove node18 travis

Not sure why it fails to install, look related to https://travis-ci.community/t/the-command-npm-config-set-spin-false-failed-and-exited-with-1-during/12909
